### ISSUE TITLE
Separate right angle indicator tolerance from angle comparison tolerances

### ISSUE THREAD

**Issue by BLFiedler** [phetsims/quadrilateral#210]
Quoting from #188, where we didn't come back to this. Deserves its own issue.

> This will NOT fix issues of overlapping tolerance such as the "looks like a rectangle/square but registers as an isosceles trap". One issue there may be that we are still using shapeAngleToleranceInterval differently. It is being used both to measure a change in angles (e.g., between adjacent angles for an isosceles trapezoid) and as a self-comparison against a static value (right angle display against 90 degrees). angleToleranceInterval is applied to a similar case re: a change in angles, but is also changed to infinity occasionally to avoid some undesirable side-dragging induced shape changes (trap -> convex, trap -> parallelogram). For now, let's create a new tolerance interval to separate the self-comparison and angle change cases. 

> - [ ] Create a new tolerance staticAngleToleranceInterval that replaces shapeAngleToleranceInterval for the case of checking 90 degrees (right angle display) or 180 degrees (convex) or anywhere else it is used to check the angle against a single value. Set the default for this to 0.01 for now. 
> - [ ] Leave shapeAngleToleranceInterval anywhere it is used for comparison between angles (e.g., corner A to corner B, between adjacent/opposite angles)
> - [ ] Create a query parameter to adjust the new tolerance interval.

> Once this is done, we may play around with the requirements for a parallelogram to address trapezoids becoming a parallelogram by dragging a side.

_Originally posted by @BLFiedler in https://github.com/phetsims/quadrilateral/issues/188#issuecomment-1227514136_

**Comment by BLFiedler** [comment]
Additional pasting from another comment: 

> So, I am also unable to break it for everything we were trying to achieve before. The one exception, once again, is the right angle indicators (make a long rectangle and then shift it just a tick to make a parallelogram OR shift two vertices one tick away to make an isoceles trapezoid).

> Located where the right angle indicators are drawn in the View
...
`assert && assert( vertex1.angleProperty.value, 'Angle must be available to draw the indicator' );
    const angle = vertex1.angleProperty.value!;
    this.visible = this.shapeModel.isRightAngle( angle ) && this.cornerGuideVisibleProperty.value;`

...
> And investigating where isRightAngle is defined in the Model and otherwise used in combination with shapeAngleToleranceInterval, led me to:

> So yeah, this is now just simply the difference between the mathematical trueness of all angles equal means 90 and the all angles equal to a tolerance derived 90.


> I am trying to determine what tolerance means in the face of the larger input grid we've discussed. It seems we've created that shape detection model that is more independent of tolerance (ignoring the right angle indicators)

> Meaning, it does exactly what we expect it to within the set tolerance.


> After playing around with extremely low to zero tolerance, I've settled on the need for the intervals for sure and now have a decent understanding of their values relative to the input grid. 

> Here's a link I used. You'll notice this doesn't work very well. Adding any nearly any level of angle tolerance fixes a few thing. I can't quite find a value less than 0.0125 for minorVertexInterval that works, but I suspect things would get worse with more interstitial values. (https://phet-dev.colorado.edu/html/quadrilateral/1.0.0-dev.61/phet/quadrilateral_en_phet.html?shapeAngleToleranceInterval=0&angleToleranceInterval=0&shapeLengthToleranceInterval=0.0001)

> And extremely large tolerances (https://phet-dev.colorado.edu/html/quadrilateral/1.0.0-dev.61/phet/quadrilateral_en_phet.html?shapeAngleToleranceInterval=5&angleToleranceInterval=5&shapeLengthToleranceInterval=10)

> This is so far, my foolproof link: [https://phet-dev.colorado.edu/html/quadrilateral/1.0.0-dev.61/phet/quadrilateral_en_phe[…]nterval=0.002&angleToleranceInterval=0.001&showModelValues](https://phet-dev.colorado.edu/html/quadrilateral/1.0.0-dev.61/phet/quadrilateral_en_phet.html?shapeAngleToleranceInterval=0.002&angleToleranceInterval=0.001&showModelValues)

**Comment by jessegreenberg** [comment]
Do we need right angle indicators still? IMO they are not as important now that we have opened the sim up as a shape detection tool. The important aspect for shape classification is equal angles (opposite and adjacent), not that there are right angles.  So if we had a visualization for equal angles, they would always match and we wouldn't have this problem at all.

So wondering if we can side-step this problem by removing or replacing the right angle indicator.

**Comment by BLFiedler** [comment]
Yeah, I believe this came up earlier and I wanted to bring it up in today's meeting, but it was already a bit topic heavy.

I do believe it's a useful measure of incremental progress toward a shape and I believe it is a specific angle we call out in Voicing independent of specific shape (in object responses?)? 

I think the use is probably more pronounced when working with a rotated shape and positioning vertices to make a square or rectangle. 

On that note, I'm leaving this unassigned for now. It may be irrelevant and we can just leave as is with the adjusted tolerances coming in #197

**Comment by BLFiedler** [comment]
Just to add some numerical evidence behind the rationale here:

I took a rectangle and translated the right side down one tick. You'll notice the top-right corner is 0.95 degrees away from 90. However, the top right and bottom right angles are 1.9 degrees away from each other (twice the value).

![image](https://user-images.githubusercontent.com/47331395/189182524-73f14832-06f4-4d3b-ae5c-97930a2261f9.png)

**Comment by BLFiedler** [comment]
I'd like to investigate quickly the behavior of angleToleranceInterval first and it may not matter practically for the chosen grid size, but for model robustness, we may consider:

- Leaving shapeAngleTolerance for any self-comparison (e.g., comparing an angle to a static number)
- Using angleToleranceInteval for any inter-angle comparison

**Comment by BLFiedler** [comment]
Discussed in meeting with @jessegreenberg. We will not do the previous comment. Instead,

- [ ] Create a new tolerance interval (and query parameter): `staticAngleToleranceInterval` which applies when comparing any angle to a number (e.g., 90 or 180 degrees). This applies to the right angle indicators and concave determination (< 180)
- [ ] Rename `shapeAngleToleranceInterval` to `interAngleToleranceInterval` to reflect its usage in comparing angles to each other (e.g., corner A to corner B). Keep all current uses except those separated in the previous bullet.
- [ ] Rename `angleToleranceInterval` to `parallelAngleToleranceInterval` to reflect its usage in determining if two opposite sides are parallel (sum of adjacent angles = 180). NOTE: it's possible this overlaps with `staticAngleToleranceInterval`, but given the nuance in parallel determination, I'd like to keep it separate in case having a different value is helpful here.

**Comment by jessegreenberg** [comment]
OK mostly done except for the following questions:

- I realized I didn't know how to apply `staticAngleToleranceInterval` to the concave check because its a check for (angle < 180) (not equality). Should this check be left alone or were you imagining it would be (angle < 180 + staticAngleToleranceInterval)?
- Should `staticAngleToleranceInterval` be "widened" when connected to a tangible device? `interAngleToleranceInterval` is, when connected to a device the value becomes `deviceInterAngleToleranceInterval` OR `deviceInterAngleToleranceInterval * toleranceIntervalScaleFactor` when shape !== null. OR can we remove all of this "widening" now? Its really complicated and maybe "Tangible Controls" in Preferences does everything we need for this.

**Comment by BLFiedler** [comment]
> * I realized I didn't know how to apply `staticAngleToleranceInterval` to the concave check because its a check for (angle < 180) (not equality). Should this check be left alone or were you imagining it would be (angle < 180 + staticAngleToleranceInterval)?

The answer to this might actually depend on if and how we decide to handle #215, since then we might need to give a little bit of wiggle room on == 180 + `staticAngleToleranceInterval`. For now though, I think you're right and we shouldn't need to apply the interval here. 

> * Should `staticAngleToleranceInterval` be "widened" when connected to a tangible device? `interAngleToleranceInterval` is, when connected to a device the value becomes `deviceInterAngleToleranceInterval` OR `deviceInterAngleToleranceInterval * toleranceIntervalScaleFactor` when shape !== null. OR can we remove all of this "widening" now? Its really complicated and maybe "Tangible Controls" in Preferences does everything we need for this.
I'll need to play with the tangible or get a better sense of this, but I agree it's a little messy. Let's chat about how we want to handle the tangible in the next meeting or the following.

For now I think this issue is good! I'll create a new issue for cleaning up the Tangible tolerance intervals and another for discussion of dynamic intervals for multitouch.

**Comment by jessegreenberg** [comment]
Thanks, the TODO pointing to this issue is about `staticAngleToleranceIntervalProperty`. @BLFiedler does it need to be "widened" like https://github.com/phetsims/quadrilateral/issues/116?

**Comment by BLFiedler** [comment]
I think we may plan to apply scaleFactor to every tolerance to adjust for tangible, multitouch, etc. Created a new issue on this topic with #247 

If it's alright to do so, feel free to remove the To Do and close this issue.

**Comment by jessegreenberg** [comment]
OK thanks, TODO removed so this issue can be closed and we will continue in #247.
